moveit_core/robot_model/include/moveit/robot_model/robot_model.h

```diff
@@ -429,10 +429,23 @@ class RobotModel
     return variable_names_;
   }
 
-  /** \brief Get the bounds for a specific variable. Throw an exception of variable is not found. */
+  /** \brief Get the bounds for a specific variable. Throw an exception of variable is not found.
+   *  \details If the variable is a mimic joint, look for the bounds of the mimicked joint variable.
+   */
   const VariableBounds& getVariableBounds(const std::string& variable) const
   {
-    return getJointOfVariable(variable)->getVariableBounds(variable);
+    const auto joint = getJointOfVariable(variable);
+    const auto mimic = joint->getMimic();
+    if (!mimic)
+    {
+      return joint->getVariableBounds(variable);
+    }
+    else
+    {
+      const auto orig_idx = joint->getLocalVariableIndex(variable);
+      const auto var_names = mimic->getVariableNames();
+      return mimic->getVariableBounds(var_names.at(orig_idx));
```

Comment:
Is this valid to do?

I didn't see an easy way in `JointModel` to know which variable in a mimic joint corresponds to which variable in its parent joint.

Reply:
This seems fine to me. It wouldn't make sense for the two mimic'd joints to have a different number of variables, and the ordering of variables should be deterministic. And even if that's not true, I've never seen a 2dof mimic joint.

Reply:
I actually ended up changing it so that the `setMimic()` function itself updates the bounds when the mimic relationship is first established.